src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -152,6 +152,34 @@ export const pinTokensWithBalanceAtTop = ({
 export const sumTokensUiUsdBalance = (tokens: TokenUi[]): number =>
 	tokens.reduce((acc, token) => acc + (token.usdBalance ?? 0), 0);
 
+/**
+ * Calculates total USD balance of mainnet tokens from the provided tokens list.
+ *
+ * @param $tokens - The list of tokens for filtering by network env and total USD balance calculation.
+ * @param $balancesStore - The balances data for the tokens.
+ * @param $exchanges - The exchange rates data for the tokens.
+ * @returns The sum of mainnet tokens USD balance.
+ *
+ */
+export const sumMainnetTokensUsdBalance = ({
+	$tokens,
+	$balances,
+	$exchanges
+}: {
+	$tokens: Token[];
+	$balances: CertifiedStoreData<BalancesData>;
+	$exchanges: ExchangesData;
+}) =>
+	$tokens.reduce((acc, token) => {
+		if (token.network.env === 'mainnet') {
```

Comment:
maybe doing a one-line definition?

`token.network.env === 'mainnet' ? acc + ..... : acc`

or something similar

Reply:
Agreed, I shortened it accordingly